core/src/main/scala/filodb.core/memstore/TimeSeriesMemStore.scala

```diff
@@ -113,11 +115,19 @@ extends MemStore with StrictLogging {
   // NOTE: Each ingestion message is a SomeData which has a RecordContainer, which can hold hundreds or thousands
   // of records each.  For this reason the object allocation of a SomeData and RecordContainer is not that bad.
   // If it turns out the batch size is small, consider using object pooling.
-  def ingestStream(dataset: DatasetRef,
-                   shardNum: Int,
-                   combinedStream: Observable[DataOrCommand],
-                   flushSched: Scheduler,
-                   diskTimeToLiveSeconds: Int): CancelableFuture[Unit] = {
+  def doIngestStream(dataset: DatasetRef,
+                     shardNum: Int,
+                     combinedStream: Observable[DataOrCommand],
+                     flushSched: Scheduler,
+                     diskTimeToLiveSeconds: Int): CancelableFuture[Unit] =
+    doIngestStream(dataset, shardNum, combinedStream, flushSched, diskTimeToLiveSeconds, Task(() => {}))
+
+  def doIngestStream(dataset: DatasetRef,
+                     shardNum: Int,
+                     combinedStream: Observable[DataOrCommand],
+                     flushSched: Scheduler,
+                     diskTimeToLiveSeconds: Int,
+                     cancelTask: Task[Unit]): CancelableFuture[Unit] = {
```

Comment:
Any reason why we could not retain one method with default arg value ?

Reply:
I was getting bombarded with compiler errors with respect to method ambiguity and I just wanted to get the thing to work. I can try again.